src/__tests__/operations/property-filter.test.ts

```diff
@@ -587,6 +587,95 @@ describe('extended operators', () => {
   });
 });
 
+describe('matching enum token', () => {
+  const obj = { value: 0 };
+  const items = [
+    /* index=0 */ { status: 'ACTIVE' },
+    /* index=1 */ { status: 'ACTIVATING' },
+    /* index=2 */ { status: 'NOT_ACTIVE' },
+    /* index=3 */ { status: 'DEACTIVATING' },
+    /* index=4 */ { status: 'TERMINATED' },
+    /* index=5 */ { status: 0 },
+    /* index=6 */ { status: obj },
+  ];
+
+  function processWithProperty(propertyKey: string, operator: string, token: any, itemsOverride = items) {
+    return processItems(
+      itemsOverride,
+      {
+        propertyFilteringQuery: { operation: 'and', tokens: [{ propertyKey, operator, value: token }] },
+      },
+      {
+        propertyFiltering: {
+          filteringProperties: [
+            {
+              key: 'status',
+              operators: [
+                { operator: '=', tokenType: 'enum' },
+                { operator: '!=', tokenType: 'enum' },
+                { operator: ':', tokenType: 'enum' },
+                { operator: '!:', tokenType: 'value' },
+              ],
+              groupValuesLabel: 'Status values',
+              propertyLabel: 'Status',
+            },
+          ],
+        },
+      }
+    ).items;
+  }
+
+  test.each(['=', '!=', ':'])('matches nothing when token=null and operator="%s"', operator => {
+    const processed = processWithProperty('status', operator, null);
+    expect(processed).toEqual([]);
+  });
+
+  test.each(['=', '!=', ':'])('matches nothing when token="" and operator="%s"', operator => {
+    const processed = processWithProperty('status', operator, '');
+    expect(processed).toEqual([]);
+  });
+
+  test('matches all when token=[] and operator="!="', () => {
+    const processed = processWithProperty('status', '!=', []);
+    expect(processed).toEqual(items);
+  });
+
+  test('matches nothing when token=[] and operator="="', () => {
+    const processed = processWithProperty('status', '=', []);
+    expect(processed).toEqual([]);
+  });
+
+  test.each([{ token: ['NOT_ACTIVE', 'ACTIVE'] }])('matches some when token=$token and operator="="', ({ token }) => {
```

Comment:
That is partially for convenience with other tests so that the tokens are defined here and not in the test body and partially to reflect the token in the test description (with $token).